Tests/GitHubEvents.tests.ps1

```diff
@@ -0,0 +1,159 @@
+# Licensed under the MIT License.
+
+<#
+.Synopsis
+   Tests for GitHubLabels.ps1 module
+#>
+
+[String] $root = Split-Path -Parent (Split-Path -Parent $Script:MyInvocation.MyCommand.Path)
+. (Join-Path -Path $root -ChildPath 'Tests\Config\Settings.ps1')
+Import-Module -Name $root -Force
+
+function Initialize-AppVeyor
+{
+<#
+    .SYNOPSIS
+        Configures the tests to run with the authentication information stored in AppVeyor
+        (if that information exists in the environment).
+
+    .DESCRIPTION
+        Configures the tests to run with the authentication information stored in AppVeyor
+        (if that information exists in the environment).
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .NOTES
+        Internal-only helper method.
+
+        The only reason this exists is so that we can leverage CodeAnalysis.SuppressMessageAttribute,
+        which can only be applied to functions.
+
+        We call this immediately after the declaration so that AppVeyor initialization can heppen
+        (if applicable).
+
+#>
+    [CmdletBinding()]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidUsingConvertToSecureStringWithPlainText", "", Justification="Needed to configure with the stored, encrypted string value in AppVeyor.")]
+    param()
+
+    if ($env:AppVeyor)
+    {
+        $secureString = $env:avAccessToken | ConvertTo-SecureString -AsPlainText -Force
+        $cred = New-Object System.Management.Automation.PSCredential "<username is ignored>", $secureString
+        Set-GitHubAuthentication -Credential $cred
+
+        $script:ownerName = $env:avOwnerName
+        $script:organizationName = $env:avOrganizationName
+
+        $message = @(
+            'This run is executed in the AppVeyor environment.',
+            'The GitHub Api Token won''t be decrypted in PR runs causing some tests to fail.',
+            '403 errors possible due to GitHub hourly limit for unauthenticated queries.',
+            'Use Set-GitHubAuthentication manually. modify the values in Tests\Config\Settings.ps1,',
+            'and run tests on your machine first.')
+        Write-Warning -Message ($message -join [Environment]::NewLine)
+    }
+}
+
+Initialize-AppVeyor
+
+$script:accessTokenConfigured = Test-GitHubAuthenticationConfigured
+if (-not $script:accessTokenConfigured)
+{
+    $message = @(
+        'GitHub API Token not defined, some of the tests will be skipped.',
+        '403 errors possible due to GitHub hourly limit for unauthenticated queries.')
+    Write-Warning -Message ($message -join [Environment]::NewLine)
+}
+
+# Backup the user's configuration before we begin, and ensure we're at a pure state before running
+# the tests.  We'll restore it at the end.
+$configFile = New-TemporaryFile
+
+try
+{
+    Backup-GitHubConfiguration -Path $configFile
+    Reset-GitHubConfiguration
+
+    if ($script:accessTokenConfigured)
```

Comment:
Why not check for this before having to touch the users' config file?  Why not exit immediately if there is no configured access token?  And why not throw an exception if that is the case?

Reply:
I am guessing that the tests were set up this way since some tests wouldn't necessarily need an accessToken to complete the tests while the ones that did would fail during PR builds? @HowardWolosky any ideas?

Reply:
That's a carry-over from the test files from before I took over the repo.  I left it in, because many r/o scenarios don't require an access token at all.  Upon re-examination, the message is slightly off (since tests aren't skipped, they will just fail as a side-effect).  We could, in theory, just immediately throw if no access token isn't provided, but again, an access token isn't strictly necessary.